crates/qt-build-utils/src/builder/qmluri.rs

```diff
@@ -0,0 +1,34 @@
+// SPDX-FileCopyrightText: 2025 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Andrew Hayzen <andrew.hayzen@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+/// A builder for representing a QML uri
+#[derive(Clone)]
+pub struct QmlUriBuilder {
+    uri: Vec<String>,
+}
+
+impl QmlUriBuilder {
+    /// Construct a [QmlUriBuilder] from a given string
+    pub fn new(uri: impl IntoIterator<Item = impl Into<String>>) -> Self {
+        Self {
+            uri: uri.into_iter().map(Into::into).collect(),
+        }
```

Comment:
This needs checks on the strings to ensure they only contain valid characters.

Unsure what exactly we want to limit it to, but to me it sounds like the whole point of such a builder is that it checks things like this for us...

Also, for now I'm happy to be more restrictive about this than potentially needed, just so we're sure we don't allow anything invalid.